Makefile

```
@@ -114,23 +115,57 @@ lib: prepare_lib
 		STACK_DEV_OPTIONS="--fast --ghc-options -Wwarn"
 
 metadata : x_token_symbol = x
-metadata : x_token_name = "Token X"
+metadata : x_token_name = Token X
 metadata : x_token_decimals = 1
 metadata : y_token_symbol = y
-metadata : y_token_name = "Token Y"
+metadata : y_token_name = Token Y
 metadata : y_token_decimals = 1
 metadata : output = metadata.json
-metadata: lib
+metadata :
+	# ============== Generate TZIP16 metadata ============== #
 	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
 		EXEC_ARGUMENTS="print-metadata \
 		--x-token-symbol $(x_token_symbol) \
-		--x-token-name $(call escape_double_quote,$(x_token_name)) \
+		--x-token-name \"$(call escape_double_quote,$(x_token_name))\" \
 		--x-token-decimals $(x_token_decimals) \
-		--y-token-symbol $(y_token_symbol)
-		--y-token-name $(call escape_double_quote,$(y_token_name)) \
+		--y-token-symbol $(y_token_symbol) \
+		--y-token-name \"$(call escape_double_quote,$(y_token_name))\" \
 		--y-token-decimals $(y_token_decimals) \
 		" EXEC_OUTPUT=$(output)
 
+metadata_byte : x_token_symbol = x
+metadata_byte : x_token_name = Token X
+metadata_byte : x_token_decimals = 1
+metadata_byte : y_token_symbol = y
+metadata_byte : y_token_name = Token Y
+metadata_byte : y_token_decimals = 1
+metadata_byte : output = metadata_byte.json
+metadata_byte :
+	# ============== Generate TZIP16 metadata in hex ============== #
+	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
+		EXEC_ARGUMENTS="print-metadata-byte \
+		--x-token-symbol $(x_token_symbol) \
+		--x-token-name \"$(call escape_double_quote,$(x_token_name))\" \
+		--x-token-decimals $(x_token_decimals) \
+		--y-token-symbol $(y_token_symbol) \
+		--y-token-name \"$(call escape_double_quote,$(y_token_name))\" \
+		--y-token-decimals $(y_token_decimals)" EXEC_OUTPUT=$(OUT)/metadata_byte.json
```

if this needs to be in a specific dir (as it seems from the `EXEC_OUTPUT`) then IMO it should not be a PHONY target:
```suggestion
$(OUT)/metadata_byte : x_token_symbol = x
$(OUT)/metadata_byte : x_token_name = Token X
$(OUT)/metadata_byte : x_token_decimals = 1
$(OUT)/metadata_byte : y_token_symbol = y
$(OUT)/metadata_byte : y_token_name = Token Y
$(OUT)/metadata_byte : y_token_decimals = 1
$(OUT)/metadata_byte:
	# ============== Generate TZIP16 metadata in hex ============== #
	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
		EXEC_ARGUMENTS="print-metadata-byte \
		--x-token-symbol $(x_token_symbol) \
		--x-token-name \"$(call escape_double_quote,$(x_token_name))\" \
		--x-token-decimals $(x_token_decimals) \
		--y-token-symbol $(y_token_symbol) \
		--y-token-name \"$(call escape_double_quote,$(y_token_name))\" \
		--y-token-decimals $(y_token_decimals)" EXEC_OUTPUT=$@
```
also, it shouldn't be a `.json` IMO given that it contains bytes in hex format.

---

I've updated this now.